iOS Client/ArticlesController/MBArticlesViewController.swift

```diff
@@ -109,73 +103,84 @@ class MBArticlesViewController: UIViewController, UITableViewDelegate, UITableVi
         case .error:
             print("Error: Loading articles")
         case .loaded(let data):
-            if data.count != articles.count {
-                print("New Data for table view")
-                articles = data
-                attributedTitles = articles.flatMap{ $0.title?.convertHtml() }
-                attributedAuthors = articles.flatMap{ $0.author?.name?.convertHtml() }
-                tableView.reloadData()
+            print("New Data for table view")
+            articlesByCategory = groupArticlesByTopLevelCategoryName(articles: data)
+            topLevelCategories = Array(articlesByCategory.keys)
+            tableView.reloadData()
+        }
+    }
+    
+    private func groupArticlesByTopLevelCategoryName(articles: [MBArticle]) -> [String: [MBArticle]] {
+        var retVal = [String: [MBArticle]]()
+        
+        articles.forEach { (article) in
+            if let categories = article.categories {
+                let topLevelCategories = Set(categories.flatMap({ (category) -> String? in
+                    return (category as? MBCategory)?.getTopLevelCategory()?.name
+                }))
+            
+                for topLevelCategory in topLevelCategories {
+                    if let _ = retVal[topLevelCategory] {
+                        retVal[topLevelCategory]?.append(article)
+                    } else {
+                        retVal[topLevelCategory] = [article]
+                    }
+                }
+                
+                if topLevelCategories.count == 0 {
```

Comment:
We might still want to display these articles some how. Not sure exactly where, probably depends on how the design ends up